crates/vm/src/compute.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    error::{ComputeError, ExecError, OpError, OpResult},
+    Access, Gas, GasLimit, LazyCache, Memory, Op, OpAccess, OpGasCost, Repeat, Stack, StateReads,
+    Vm,
+};
+use rayon::prelude::*;
+use std::sync::Arc;
+
+/// The limit on compute recursion depth.
+pub const MAX_COMPUTE_DEPTH: usize = 1;
```

Comment:
Should this be higher?

Reply:
Good question, would that be useful for anything?

Reply:
I think we should just make this something that won't surprise people like `10`. A depth of `1` means you can't have a compute within a compute at all.